source/guides/network/firewall.rst

```diff
@@ -0,0 +1,204 @@
+      sudo iptables -A INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+
+
+#. Test the running firewall configuration to make sure it behaving as
+   you expect. 
+
+#. Run the *iptables-save* service to make the running configuration
+   persistent. This will perform a one-time save of the running configuration
+   to :file:`/etc/iptables.rules` :
+
+   .. code:: bash
+
+      sudo systemctl start iptables-save
+
+
+
+
+#. Enable the *iptables-resolve* service the iptables rules to be
+   automatically applied at boot from the :file:`/etc/iptables.rules` file:
+
+   .. code :: bash
+
+      sudo systemctl enable iptables-restore.service
+
+
+ipset
+=====
+
+`ipset <http://ipset.netfilter.org/>`_ is a framework in the Linux kernel for
+storing and efficiently indexing combinations of IP addresses, networks,
+(TCP/UDP) port numbers, MAC addresses, and interface names. 
+
+IP sets makes writing network policy rules simpler and processing them against
+a large and/or changing sets of hosts more efficient. 
+
+By themselves, IP sets do not enforce network traffic rules but can be used to
+extend iptables rules for matching. It is important to note that the ipset
+must be defined before a netfilter rule can match against it.
+
+* Running IP sets can be manipulated with the :command:`ipset` utility. 
+
+* Custom IP sets can be stored in the :file:`/etc/ipset.conf` file
+
+* IP sets in :file:`/etc/ipset.conf` can be automatically applied at boot by
+  enabling the *ipset* service with the command :command:`sudo systemctl
+  enable ipset`.
+
+See :command:`man ipset` to learn more about using ipsets. 
+
+
+firewalld
+=========
+
+`firewalld <https://firewalld.org/>`_ is based on nftables, the successor to
+iptables and parts of the netfilter framework. The description of firewalld
+helps highlight some of the differences compared to iptables:
+
+    firewalld provides a dynamically managed firewall with support for
+    network/firewall zones to define the trust level of network connections or
+    interfaces. It has support for IPv4, IPv6 firewall settings and for
+    ethernet bridges and has a separation of runtime and permanent
+    configuration options. It also supports an interface for services or
+    applications to add firewall rules directly.
+
+See :command:`man firewalld` for more information. 
+
+Below is information on using :command:`firewalld` on |CL|: 
+
+#. Install he *firewalld* bundle:
+
+   .. code:: bash
+
+      sudo swupd bundle-add firewalld
+
+
+#. Disable *iptables* and *ipset* services as they conflict with firewalld:
+
+   .. code:: 
+
+      sudo systemctl mask iptables-restore ipset
+
+
+#. :command:`firewall-cmd` can be used to configure the running or permanent
+   firewall configuration. See the `firewalld documentation
+   <https://firewalld.org/documentation/>`_ to learn more about
+   firewalld concepts and configuration options.
+
+   Below is a common example to allow HTTP traffic in public zones:
```

Comment:
We should default to `HTTPS` here for the example. Not insecure HTTP.

Reply:
Waiting on final edit. Will merge after your requested change is made.

Reply:
Done.

Reply:
Cheers.